hdk/virtual_machine/code.py

```diff
@@ -31,6 +33,173 @@
+
+
+def translate_branching_command(command: BranchingCommand) -> list[str]:
+    """Translates a BranchingCommand into its assembly code.
+
+    Args:
+        command: The BranchingCommand to translate.
+
+    Returns:
+         A list of assembly code instructions.
+    """
+    if command.operation == "label":
+        return translate_label(command.label_name)
+    if command.operation == "goto":
+        return translate_goto(command.label_name)
+    if command.operation == "if-goto":
+        return translate_if_goto(command.label_name)
+    raise ValueError(f"Wrong operation {command.operation!r} for branching command.")
+
+
+def translate_call(
+    function_name: str, n_args: int | None, command_id: int
+) -> list[str]:
+    """Translates a call command into its assembly code.
+
+    Args:
+        function_name: The symbol defined by the function.
+        n_args: The number of arguments.
+        command_id: The current command number.
+
+    Returns:
+        A list of assembly code instructions.
+    """
+    output = [
+        f"@{function_name}{command_id}",
+        "D=A",
+        "@SP",
+        "A=M",
+        "M=D",
+        "@SP",
+        "M=M+1",
+    ]
+    for segment in ["LCL", "ARG", "THIS", "THAT"]:
+        output += [f"@{segment}", "D=M", "@SP", "A=M", "M=D", "@SP", "M=M+1"]
+    output += [
+        "@SP",
+        "D=M",
+        "@5",
+        "D=D-A",
+        f"@{n_args}",
+        "D=D-A",
+        "@ARG",
+        "M=D",
+        "@SP",
+        "D=M",
+        "@LCL",
+        "M=D",
+        f"@{function_name}",
+        "0;JMP",
+        f"({function_name}{command_id})",
+    ]
+    return output
+
+
+def translate_return():
+    """Translates a return command into its assembly code.
+
+    Returns:
+        A list of assembly code instructions.
+    """
+    output = [
+        "@LCL",
+        "D=M",
+        "@R13",
+        "M=D",
+        "@5",
+        "A=D-A",
+        "D=M",
+        "@R14",
+        "M=D",
+        "@SP",
+        "AM=M-1",
+        "D=M",
+        "@ARG",
+        "A=M",
+        "M=D",
+        "@ARG",
+        "D=M+1",
+        "@SP",
+        "M=D",
+    ]
+    for seg in ["THAT", "THIS", "ARG", "LCL"]:
+        output += ["@R13", "AM=M-1", "D=M", f"@{seg}", "M=D"]
+    return output + ["@R14", "A=M", "0;JMP"]
+
+
+def translate_function(function_name: str, n_vars: int | None) -> list[str]:
+    """Translates a function command into its assembly code.
+
+    Args:
+        function_name: The symbol defined by the function.
+        n_vars: The number of the function's variables.
+
+    Returns:
+         A list of assembly code instructions.
+    """
+    initialize_locals = []
+    for _ in range(n_vars if n_vars is not None else 0):
+        initialize_locals += translate_push("constant", 0) + _SAVE_D_INSTRUCTIONS
+    return [f"({function_name})"] + initialize_locals
+
+
+def translate_function_command(command: FunctionCommand, command_id) -> list[str]:
```

Comment:
Remove this function, because instead of FunctionCommand you will have 3 different commands.